dlgr/griduniverse/experiment.py

```diff
@@ -1133,27 +1201,44 @@ def __init__(self, session=None):
             self.setup()
             self.grid = Gridworld(
                 log_event=self.record_event,
-                **self.config.as_dict()
+                item_config=self.item_config,
+                transition_config=self.transition_config,
+                **self.config.as_dict(),
             )
             self.session.commit()
 
     def configure(self):
         super(Griduniverse, self).configure()
-        self.num_participants = self.config.get('max_participants', 3)
+        self.num_participants = self.config.get("max_participants", 3)
         self.quorum = self.num_participants
-        self.initial_recruitment_size = self.config.get('num_recruits',
-                                                        self.num_participants)
-        self.network_factory = self.config.get('network', 'FullyConnected')
+        self.initial_recruitment_size = self.config.get(
+            "num_recruits", self.num_participants
+        )
+        self.network_factory = self.config.get("network", "FullyConnected")
 
         game_config_file = os.path.join(os.path.dirname(__file__), GAME_CONFIG_FILE)
-        with open(game_config_file, 'r') as game_config_stream:
+        with open(game_config_file, "r") as game_config_stream:
             self.game_config = yaml.safe_load(game_config_stream)
-        self.object_config = {
-            o['object_id']: o for o in self.game_config.get('objects', ())
-        }
+        self.item_config = {o["item_id"]: o for o in self.game_config.get("items", ())}
+
+        # If any item is missing a key, add it with default value.
+        item_defaults = self.game_config.get("item_defaults", {})
+        for item in self.item_config.values():
+            for prop in item_defaults:
+                if prop not in item:
+                    item[prop] = item_defaults[prop]
+
         self.transition_config = {
-            (t['actor_start'], t['target_start']): t for t in self.game_config.get('transitions', ())
+            (t["actor_start"], t["target_start"]): t
+            for t in self.game_config.get("transitions", ())
         }
+        # This is accessed by the grid.html template to load the configuration on the client side:
+        # TODO: could this instead be passed as an arg to the template in
+        # the /grid route?
+        self.item_config_json = json.dumps(self.item_config)
```

Comment:
Seems a little redundant to store this on the class. I think the idea of initializing the template with it or providing it via some initialize JSON route.